File: Factorial/Factorial.py
###
# Factorial.py
# by Daniel Brice
# Invoke as `python3 Factorial.py <num>`
# Prints <num> factorial
###
import sys

def factorial(n):
  if n is 0:
    return 1
  else:
    return n * factorial(n-1)

print(factorial(int(sys.argv[1])))

File: Factorial/Factorial.scala
/**
 * Factorial.scala
 * by Daniel Brice
 * Invoke as `scala Factorial <num>`
 * Prints <num> factorial
 */

object Factorial {

  def factorial(n: Int): Int = {
    1.to(n).foldLeft(1)(_ * _)
  }

  def main(args: Array[String]): Unit = {
    println(factorial(args(0).toInt))
  }
}
